test-clients/src/main/java/com/hedera/services/bdd/junit/ExpiryRecordsValidator.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (C) 2022-2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hedera.services.bdd.junit;
+
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.SUCCESS;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import com.hederahashgraph.api.proto.java.TransactionRecord;
+import java.util.ArrayList;
+import java.util.List;
+
+/** This validator validates expiry contract records at the end of test run */
+public class ExpiryRecordsValidator implements RecordStreamValidator {
+    private static final String AUTO_RENEWAL_MEMO = " was automatically renewed";
+    private static final String AUTO_EXPIRY_MEMO = " was automatically deleted";
+
```

Comment:
Open to any suggestions on any other validations that  can be added.

Reply:
Would it be helpful to validate that the sum total of the transfers is 0? That's the only thing I can think of